tephra-hbase-compat-0.96/src/main/java/co/cask/tephra/hbase96/coprocessor/TransactionProcessor.java

```diff
@@ -149,10 +149,13 @@ public void preGetOp(ObserverContext<RegionCoprocessorEnvironment> e, Get get, L
     throws IOException {
     Transaction tx = getFromOperation(get);
     if (tx != null) {
+      boolean excludeCurrentWritePtr = isAttributeSet(get, TxConstants.TX_EXCLUDE_CURRENT_WRITE);
```

Comment:
See my other comments. I think it is better if data operations are completely unaware of whether the transaction excludes the current write version or not. They should simply rely on tx.isVisible() to determine whether a version is visible. That limits the change to the Transaction class itself, and minimizes the change required in every transaction-aware, to be compatible with checkpoints.